tests/interpolation/test_custom_resolvers.py

```diff
@@ -38,6 +59,30 @@ def signature_not_inspectable(*args: Any, **kw: Any) -> None:
     assert OmegaConf.create({"x": "${not_inspectable:}"}).x == 123
 
 
+@mark.parametrize(
+    ("use_cache_1", "use_cache_2", "expected"),
+    [
+        (False, False, 2),
+        (False, True, 2),
+        (True, False, 2),
+        (True, True, 1),  # potentially controversial, see #637
```

Comment:
Commented. I think this is okay and you can change the comment to indicate that we are getting the value from the cache.
Alternatively you can use a new config object or clear the cache on the existing one.

Reply:
Done in cbe3a8e (I prefer to keep the test in case this would change in the future)